apps/settings/lib/Controller/CheckSetupController.php

```diff
@@ -524,8 +524,12 @@ protected function getOpcacheSetupRecommendations(): array {
 			}
 
 			if (
-				empty($status['interned_strings_usage']['free_memory']) ||
-				($status['interned_strings_usage']['used_memory'] / $status['interned_strings_usage']['free_memory'] > 9)
+				// Do not display this error if the buffer_size is already set to a firth of the total opcache memory consumption or higher
+				(!$status['interned_strings_usage']['buffer_size'] >= (($this->iniGetWrapper->getNumeric('opcache.memory_consumption') / 4) * 1024 * 1024)) && 
+				(
+					empty($status['interned_strings_usage']['free_memory']) ||
+					($status['interned_strings_usage']['used_memory'] / $status['interned_strings_usage']['free_memory'] > 9)
+				)
```

Comment:
The value is empty when the interned strings buffer is set to `0`, in which case we should recommend to enable it and avoid a syntax error. Also we need to compare the setting's value instead of the `opcache_get_status` one, since the returned interned strings buffer size (as well as usage stats) is only 75% of the assigned value, hence cannot be compared directly. With `/ 3` it would be possible then, but it is even more complicated if values are assigned which are no power of 2: https://stackoverflow.com/q/67853338/16145737
```suggestion
				empty($status['interned_strings_usage']['free_memory']) ||
				(
					($status['interned_strings_usage']['used_memory'] / $status['interned_strings_usage']['free_memory'] > 9) &&
					// Do not recommend to raise the interned strings buffer size above a quarter of the total OPcache size
					($this->iniGetWrapper->getNumeric('opcache.interned_strings_buffer') < $this->iniGetWrapper->getNumeric('opcache.memory_consumption') / 4)
				)
```

Reply:
done! thanks for the review! :)